src/utils/kv_map.rs

```diff
@@ -180,6 +207,71 @@ impl<K: Clone + Ord, V: Clone> IntoIterator for RecordingMap<K, V> {
     }
 }
 
+// KV MAP DIFF
+// ================================================================================================
+/// [KvMapDiff] stores the difference between two key-value maps.
+///
+/// The [KvMapDiff] is composed of two parts:
+/// - `updates` - a map of key-value pairs that were updated in the second map compared to the
+///               first map. This includes new key-value pairs.
+/// - `removed` - a set of keys that were removed from the second map compared to the first map.
+#[derive(Debug, Clone)]
+pub struct KvMapDiff<K, V> {
+    updates: BTreeMap<K, V>,
+    removed: BTreeSet<K>,
+}
```

Comment:
minor nit: should we rename `updates` to `updated` to keep the tenses consistent with `removed`?